File: 94_Binary_Tree_Inorder_Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        result=[]
        if not root:
            return result
        def helper(node):
            if(node):
                if(node.left):
                    helper(node.left)
                result.append(node.val)
                if(node.right):
                    helper(node.right)
                
        helper(root)
        return result
                
            
        

File: 94_Binary_Tree_Inorder_Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    
    void Traversal(TreeNode* root, vector<int> &v)
    {
        if(root != NULL)
        {
            Traversal(root->left,v);
            v.push_back(root->val);
            Traversal(root->right,v);
        }
    }
    
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> v;
        
        Traversal(root,v);
        
        return v;
    }
};